test/provision/container-images.sh

```diff
@@ -21,8 +21,13 @@ function check_img_list {
 
 function test_images {
   echo "Downloading all container images needed for tests"
-  DOCKER_IMAGES=$(grep -rI "docker.io/.*/.*:.*" test/ | sed -e 's/\"//g' | grep -v "{}"  | sed -e 's/.*docker.io/docker.io/g' |  sort | uniq)
-  QUAY_IMAGES=$(grep -rI "quay.io/.*/.*:.*" test/ | sed -e 's/\"//g' | grep -v "{}" | sed -e 's/.*quay.io/quay.io/g'  | sort | uniq)
+  # Filter out just image names.
+  # grep's -I ignores binary files, -no-filename doesn't print filenames
+  # sed's -n does not print non-matching lines and the `#gp` prints only
+  # matching groups, and the match is global. `#` is used as the sed delimiter
+  # to avoid escaping `/` in the regex.
```

Comment:
:+1: :+1: :+1: 

Useful explanation. Only thing is, where's the `#gp`, I only see `#p`?

Reply:
ah, I had the global in an earlier version but that proved unnecessary. I've cleaned it up!